### PR TITLE
release GlobalMax/MeanPool1/2d

### PR DESCRIPTION
**Pull request by zsdonghao**
Support Global Max/Mean Pool 1/2d

### CODE REVIEW

File: tests/test_pooling.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import tensorflow as tf
+from tensorlayer.layers import *
```
Comment (lgarithm) [inline tests/test_pooling.py]: don't use import * for new code.

File: tensorlayer/layers/pooling.py [hunk 2]
```diff
@@ -249,19 +266,214 @@ def meanpool3d(net, filter_size=(3, 3, 3), strides=(2, 2, 2), padding='valid', d
...
+        self.all_params = list(layer.all_params)
+        self.all_drop = dict(layer.all_drop)
+
+        # update layer (customized)
+        self.all_layers.append(self.outputs)
+        # self.all_params.extend( [W, b] )
+
+
+class GlobalMeanPool1d(Layer):
+    """ The :class:`GlobalMeanPool1d` class is a 1D Global Mean Pooling layer.
```
Comment (lgarithm) [inline tensorlayer/layers/pooling.py]: No space after `"""`

File: tests/test_pooling.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import tensorflow as tf
```
Comment (lgarithm) [inline tests/test_pooling.py]: need to add this file to `.travis-ci.yml`

File: tensorlayer/layers/pooling.py [hunk 4]
```diff
@@ -249,19 +266,214 @@ def meanpool3d(net, filter_size=(3, 3, 3), strides=(2, 2, 2), padding='valid', d
...
+        # the input of this layer is the output of previous layer (fixed)
+        self.inputs = layer.outputs
+
+        # print out info (customized)
+        logging.info("GlobalMeanPool2d %s" % name)
+
+        # operation (customized)
+        self.outputs = tf.reduce_mean(layer.outputs, axis=[1, 2], name=name)
+
+        # get stuff from previous layer (fixed)
```
Comment (lgarithm) [inline tensorlayer/layers/pooling.py]: @luomai shall we consider slowly move this into the `Layer` base class?

```
 self.all_layers = list(layer.all_layers)
 self.all_params = list(layer.all_params)
 self.all_drop = dict(layer.all_drop)
```

Comment (lgarithm) [reply tensorlayer/layers/pooling.py]: in `ALayer(Layer):`
        Layer.__init__(self, name=name, layer=layer)
in `Layer`:
```
    class Layer(object):
        def __init__(self, name, layer):
             self.all_layers = list(layer.all_layers)
            self.all_params = list(layer.all_params)
            self.all_drop = dict(layer.all_drop)
```

so we don't have to do it every layer.

Comment (lgarithm) [reply tensorlayer/layers/pooling.py]: We can change it slowly by:
1. introduce `layer` argument to `__init__` of `Layer` as optional.
2. slowly make all sub layers pass the layer argument to `Layer.__init___`
3. make `layer` required in `__init__`

Comment (zsdonghao) [reply tensorlayer/layers/pooling.py]: we can do this in another repo?

Comment (lgarithm) [reply tensorlayer/layers/pooling.py]: Of course, this needs a careful plan before carrying out.

**Pull request merged**